File: striver/S3L1/rotated_sort_check.java
package S3L1;

public class rotated_sort_check {
    public boolean check(int[] arr) {
        int count=0;
        int pin = 0;
        for(int i=0;i<arr.length-1;i++){
                if(arr[i]>arr[i+1]) {
                    pin = i;
                    count++;
                }
            }   
        if(count==0)return true;
        if(count==1) {
            for(int i=pin+1;i<pin+arr.length;i++){
                if(arr[i%arr.length]>arr[(i+1)%arr.length]) {
                    return false;
                }
            }
            return true;
        }
        return false;
    }
}

File: striver/S3L1/rotated_sort_check.py
class Solution(object):
    def check(self, nums):
        count=0
        pin = 0
        for i in range(0,len(nums)-1):
            if nums[i]>nums[i+1]:
                pin = i
                count+=1
        if count==0 :
            return True
        if count==1:
            for i in range(pin+1,pin+len(nums)):
                if nums[i%len(nums)]>nums[(i+1)%len(nums)] :
                    return False
            return True
        return False
       